### PR TITLE
uid is also supported in field ref for downward api

### PR DESCRIPTION
**Pull request by kannon92**
#### What type of PR is this?
/kind cleanup
/kind documentation

#### What this PR does / why we need it:
UID is supported for downward api in pkg/apis but the staging comment does not match.

https://kubernetes.io/docs/concepts/workloads/pods/downward-api/#downwardapi-fieldRef

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/api/core/v1/types.go [hunk 1]
```diff
@@ -7093,7 +7093,7 @@ const (
 type DownwardAPIVolumeFile struct {
 	// Required: Path is  the relative path name of the file to be created. Must not be absolute or contain the '..' path. Must be utf-8 encoded. The first item of the relative path must not start with '..'
 	Path string `json:"path" protobuf:"bytes,1,opt,name=path"`
-	// Required: Selects a field of the pod: only annotations, labels, name and namespace are supported.
+	// Required: Selects a field of the pod: only annotations, labels, name, namespace and uid are supported.
```
Comment (sftim) [inline staging/src/k8s.io/api/core/v1/types.go:7096]: This is less wrong, but https://kubernetes.io/docs/concepts/workloads/pods/downward-api/ lists other fields you could select.

Comment (sftim) [reply staging/src/k8s.io/api/core/v1/types.go:7096]: If we want a short URL for supported fields, we can make one.

Comment (kannon92) [reply staging/src/k8s.io/api/core/v1/types.go:7096]: That API says we support those but not in FieldRef. 

DownwardAPI supports them as a few options but we directly expose FieldRef in uuid. That comment is actually in pkg/api/core so wanted to make sure they were identical.

And for "supported fields" are you saying that we should just link to the code comment?

Comment (sftim) [reply staging/src/k8s.io/api/core/v1/types.go:7096]: Ah, yep, that's right. My mistake.

**Comment by liggitt** [comment]
I was a little surprised to see PRs opened without much discussion on expanding the downward API surface area.

@thockin, looks like you commented on the creationTimestamp one. Were you expecting these to just be added ad-hoc?

Dropping this out of the api-review queue until it's clearer the principles for expanding this.

cc @jpbetz @deads2k for api-machinery visibility

/remove-label api-review

**Comment by kannon92** [comment]
@liggitt this PR is mostly cleaning up documentation between staging and pkg/apis. We already support uid. It wasn't clear in the staging docs that we do. This PR is mostly just adding that to staging.

**Comment by kannon92** [comment]
https://github.com/kubernetes/kubernetes/blob/master/pkg/apis/core/types.go#L1424

This says we support uid but the comment did not match for staging so I think this should be fine as is.

**Comment by liggitt** [comment]
oh, ha... I only read the PR title and not the change 🤦 

/lgtm
/approve

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Comment by pacoxu** [comment]
The UT failed for https://github.com/kubernetes/kubernetes/pull/123603

/hold

**Pull request merged**